app.rb

```diff
@@ -0,0 +1,119 @@
+module Enumerable
+  def my_each
+    return enum_for unless block_given?
+
+    i = 0
+    new_array = self
+    new_array = to_a if is_a?(Range) || is_a?(Hash)
+    while i < new_array.length
+      yield new_array[i]
+      i += 1
+    end
+    new_array
+  end
+
+  def my_each_with_index
+    return enum_for unless block_given?
+
+    i = 0
+    new_array = self
+    new_array = to_a if is_a?(Range) || is_a?(Hash)
+    while i < new_array.length
+      yield new_array[i], i
+      i += 1
+    end
+    new_array
+  end
+
+  def my_select
+    return enum_for unless block_given?
+
+    new_numbers = []
+    my_each { |i| new_numbers.push(i) if yield(i) }
+    new_numbers
+  end
+
+  def my_all?(arg = nil)
+    return false if arg.nil? && !block_given? && !my_each { |j| j }
+
+    unless arg.nil?
+      return false unless my_each { |j| j.class == arg || arg.match(j) }
+    end
+    return false if block_given? && !my_each { |j| yield(j) }
+
+    true
+  end
```

Comment:
# my_all?
1. [ ]  Returns false if the block returns false or nil
### Test 👉 `array.my_all?(&false_block)`
|Expected|Got|
|:---:|:---:|
|`false`|`true`|
#
2. [ ] Returns false if the block returns false or nil::range
### Test 👉 `range.my_all?(&false_block)`
|Expected|Got|
|:---:|:---:|
|`false`|`true`|
#
3. [ ] When no block or argument is given returns false when one of the collection members is false or nil
### Test 👉 `false_array.my_all?`
|Expected|Got|
|:---:|:---:|
|`false`|`true`|
#
4. [ ] When a class is passed as an argument returns true if all of the collection is a member of such class:: Numeric
### Test 👉 `numbers.my_all?(Numeric)`
Resulted in *NoMethodError* 
# 
5. [ ] When a class is passed as an argument returns false if any of the collection is not a member of such class
### Test 👉 `array.my_all?(Integer)`
Resulted in *NoMethodError* 
#
6. [ ]  When a Regex is passed as an argument returns false if any of the collection does not match the Regex
### Test 👉 `words.my_all?(/o/)`
|Expected|Got|
|:---:|:---:|
|`false`|`true`|
#
7.  When a pattern other than Regex or a Class is given returns true if all of the collection matches the pattern
### Test 👉 `array.my_all?(3)`
Resulted in *NoMethodError*
#
8. [ ] When a pattern other than Regex or a Class is given returns false if any of the collection does not match the pattern
### Test 👉 `array.my_all?(3)`
Resulted in *NoMethodError*